amads/melody/meteraccent.py

```diff
@@ -0,0 +1,193 @@
+__author__ = "Yiwen Zhao"
+
+import numpy as np
+from ..core.basics import Score, Note, Part
+
+def calculate_meteraccent(score: Score) -> float:
+    """
+    Calculate the phenomenal accent synchrony in a melody.
+    
+    This function measures how well different types of musical accents
+    (durational, pitch, and metrical) align with each other. Higher
+    synchrony indicates stronger alignment between different accent types.
+    
+    Parameters
+    ----------
+    score : Score
+        A Score object containing the melody to analyze. The score will be
+        flattened and collapsed into a single sequence of notes ordered by
+        onset time.
+    
+    Returns
+    -------
+    float
+        Accent synchrony value between 0 and 1.
+        Higher values indicate greater synchronization between different
+        types of accents.
+        Returns 0 for empty scores or single notes.
+    
+    References
+    ----------
+    .. [1] Eerola, T., Himberg, T., Toiviainen, P., & Louhivuori, J. 
+           Perceived complexity of Western and African folk melodies by 
+           Western and African listeners.
+    .. [2] Jones, M. R. (1987). Dynamic pattern structure in music: Recent 
+           theory and research. Perception and Psychophysics, 41, 621-634.
+    
+    Examples
+    --------
+    >>> score = Score()
+    >>> part = Part()
+    >>> notes = [
+    ...     Note(pitch=60, dur=1.0),  # Quarter note C4
+    ...     Note(pitch=64, dur=0.5),  # Eighth note E4
+    ...     Note(pitch=67, dur=1.0)   # Quarter note G4
+    ... ]
+    >>> for note in notes:
+    ...     part.append(note)
+    >>> score.append(part)
+    >>> accent_sync = calculate_meteraccent(score)
+    >>> print(accent_sync)
+    0.75
+    """
+    def calculate_durational_accents(notes: list) -> np.ndarray:
+        """Calculate durational accents based on relative note durations."""
+        durations = np.array([note.dur for note in notes])
+        # Normalize durations to create accent weights
+        return durations / np.max(durations)
+    
+    def calculate_pitch_accents(notes: list) -> np.ndarray:
+        """Calculate pitch accents based on melodic intervals."""
+        pitches = np.array([note.keynum for note in notes])
+        intervals = np.abs(np.diff(pitches))
+        # Pad first note to match length
+        intervals = np.pad(intervals, (1, 0), 'constant')
+        # Normalize intervals to create accent weights
+        return intervals / np.max(intervals) if np.max(intervals) > 0 else np.zeros_like(intervals)
+    
+    def calculate_metrical_accents(notes: list) -> np.ndarray:
+        """Calculate metrical accents based on metric position."""
+        # Assuming 4/4 meter for simplicity
+        # Could be extended to handle different time signatures
+        METER_PATTERN = np.array([1.0, 0.2, 0.5, 0.2])  # 4/4 meter weights
```

Comment:
Let's call these `meter_weights` or similar please. Pattern is ambiguous.

Reply:
Worth moving this to an argument too. We will certainly offer different profiles.

Maybe also worth a note to say that this argument type may change.